core-processor/src/ext.rs

```diff
@@ -690,6 +690,10 @@ impl EnvExt for Ext {
             return Err(ReservationError::ZeroReservationDuration.into());
         }
 
+        if amount < self.context.mailbox_threshold {
```

Comment:
I'd remove check `amount == 0` from above and mark ZeroReservationAmount err variant as `#[deprecated]` (because we should support backward compatibility)